routing/matcher.go

```diff
@@ -326,7 +315,8 @@ func newMatcher(rs []*Route, o MatchingOptions) (*matcher, []*definitionError) {
 		}
 
 		if r.pathSubtree != "" {
-			path, _ := cleanPath(r.pathSubtree, o|IgnoreTrailingSlash)
+			path := httppath.Clean(r.pathSubtree)
+			path = trimTrailingSlash(path)
```

Comment:
don't we need to use the `if o.ignoreTrailingSlash()` condition here?

Reply:
oops.. my bad for not watching :man_facepalming:

Reply:
`(o|IgnoreTrailingSlash).ignoreTrailingSlash()` is always true